### PR TITLE
improve dockerfile

### PR DESCRIPTION
**Pull request by joschahenningsen**
This pr makes use of multi stage building to separate dependency and code changes.

Currently rebuilding the image takes about 10 minutes on my laptop even for tiny code changes because all dependencies have to be reinstalled. This PR makes the dependency stuff cache-able by docker so every subsequent build takes only a couple of seconds.

### CODE REVIEW

**Comment by joschahenningsen** [comment]
Also this should help with this :D

```sh
$ docker system prune
> Total reclaimed space: 31.14GB
```

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,22 @@
+FROM python:3.9-slim as builder
+
+ADD requirements.txt .
+
+# Whisper Dependencies
+RUN apt-get update && apt-get install -y git
+
+# Install Python modules
+RUN pip install --user --no-cache-dir -r requirements.txt
```
Comment (CommanderStorm) [inline Dockerfile]:
```suggestion
# Dependencys
RUN apt-get update && apt-get install -y git \
    && pip install --user --no-cache-dir -r requirements.txt 
```
Installing this in one command is better, as this way the `builder` takes up less space.
If cachability of docker builds is a concern, the whisper dependencys can be moved above the `requirements.txt`

**Review by CommanderStorm — COMMENTED** [review]
just a small thing I noticed

File: Dockerfile (hunk 1 above)
Comment (joschahenningsen) [reply Dockerfile]: Good catch!

File: Dockerfile [hunk 2]
```diff
@@ -1,19 +1,20 @@
+FROM python:3.9-slim as builder
+
+ADD requirements.txt .
+
+# Dependencys
```
Comment (MatthiasReumann) [inline Dockerfile]:
```suggestion
# Dependencies
```

**Review by MatthiasReumann — APPROVED** [review]
Looks good. 💯

File: Dockerfile (hunk 1 above)
Comment (MatthiasReumann) [reply Dockerfile]: Why is that actually? In other words, why does the builder take less space if you run this commands as a single one using `&&`?

Comment (CommanderStorm) [reply Dockerfile]: Here is a Stack Overflow post which explains this in detail: https://stackoverflow.com/questions/39223249/multiple-run-vs-single-chained-run-in-dockerfile-which-is-better

TLDR: likely is not significant (both negative+positive) in our use case.
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers

Comment (CommanderStorm) [reply Dockerfile]: Ahh.. Seems this answer is outdated: https://stackoverflow.com/a/47451019
This change can be reverted. Sorry
